### PR TITLE
Updates for IETF 116

### LINKED ISSUE #125: Change the name of nmc-path-id list
We need to change the name of nmc-path-id , with e2e-mc-path-id and related description.

### LINKED ISSUE #126: OI topology profile
Based on previous discussion (see [minutes-2022-11-15](https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/master/minutes/minutes-2022-11-15.md)), an issue has been raised about the possibility to use a sub-set of the OI topology model for other applications.

One option is to profile the model (like in [draft-busi-teas-te-topology-profiles](https://datatracker.ietf.org/doc/html/draft-busi-teas-te-topology-profiles)).

However, there are attributes which are mandatory for OI applications and cannot be profiled in the current YANG model.

Italo, Sergio, Dieter, Aihua, Julien and Esther had an offline meeting to explore this possibility and it seems possibile to update the YANG model to allow profiling.

However, there is a need to indicate somewhere within the draft (e.g., within the YANG description statements) which attributes are mandatory for OI applications to have a multi-vendor profile for OI applications.

This is the summary of the proposed changes to YANG to allow profiling the OI topology model to support other applications beside the OI applications:

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L6-L11

    How to interpret an empty list?
    
    a) no OTSis are configured on the link?
    b) OTSis configured on the link are not reported (profile)
    
    **Proposal**: use a presence container to indicate whether OTSis are reported
    
    NEW
    ```
       +--ro otsis!
          +--ro otsi-group* [otsi-group-id]
             +--ro otsi-group-id    string
             +--ro otsi* [otsi-carrier-id]
                +--ro otsi-carrier-id           uint16
                +--ro otsi-carrier-frequency?   union
                +--ro nmc-path-id*              uint16
    ```

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L13

    How to interpret an empty list?
    
    **Proposal**: use a presence container to indicate whether transponders are reported
    
    NEW
    ```
        +--ro transponders!
          +--ro transponder* [transponder-id]    
    ```

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L17

    When the server report the transponders it shall also report the transceivers within the transponder and a transponder shall have at least one transceiver.

    **Proposal**: specify a min-element of 1 (a transponder shall contain at least one transceiver).

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L19-L20

    No ambiguity with an empty list since a transceiver will support at least one mode.
    
    Two options:
    a) make the supported-modes a presence container and add a min-element 1 to the supported-mode*
    b) leave as it is clarifying in the description that the list is empty  when the server does not report the supported-modes (profile)

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L114-L115

    How to interpret when this attribute is not present?
    
    a) the transceiver is not configured (and there is no default configuration)
    b) the configured-mode is not reported by the server (profile)
    
    **Proposal**: define as a union to indicate that no mode is configured:
    - a NULL value is reported if no mode is configured and there is no default mode;
    - if there is a default mode, the attribute reports the default unless a different mode is configured;
    - if not present, it means that the configured-mode is not reported by the server (profile).

1. - [ ] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L154

    How to interpret an empty list?
    
    **Proposal**: use a presence container to indicate whether OMS-elements are reported
    
    ```
           +--ro OMS-elements!
              +--ro OMS-element* [elt-index]
    ```
    
    No need to profile a fine grain the attributes within the OMS-element list since if implemented it is implemented for performing path computation in the client controller: what is mandatory within this list can be kept as mandatory (since it is mandatory only when the OMS-elements container is present)

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L119-L126

    How to interpret an empty container?
    
    1) no incoming/outgoing OTSi has been setup from this transceiver (unused transceiver)
    2) OTSi information is not reported by the server (profile)
    
    **Proposal**: use a presence container to indicate whether OTSis are reported
    
    NEW
    ```
        |     +--ro outgoing-otsi!
        |     |  +--ro otsi-group-ref?
        |     |  |       -> ../../../../../otsi-group/otsi-group-id
        |     |  +--ro otsi-ref?         leafref
        |     +--ro incoming-otsi!
        |     |  +--ro otsi-group-ref?
        |     |  |       -> ../../../../../otsi-group/otsi-group-id
        |     |  +--ro otsi-ref?         leafref
    ```

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L127

    How to interpret when this attribute is not present?
    
    a) no termination type has been configured (unused transceiver)
    b) termination type information is not reported by the server (profile)
    
    **Proposal**: add a value 'unused-transceiver' to disambiguate:
    - if 'unused-transceiver' it means that the associated transceiver is not used;
    - if not reported it means that the server does not report this information (profile)

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L128-L131

    How to interpret an empty list?
    
    **Proposal**: use a presence container to indicate whether OMS-elements are reported:
    
    NEW
    ```
        +--ro regen-groups!
           +--ro regen-group* [group-id]
              +--ro group-id           uint32
              +--ro regen-metric?      uint32
              +--ro transponder-ref*   -> ../../transponder/transponder-id
    
    ```
    For the transponder-ref: add a min-element 1 and a when statement indicating that the list applies only when the  transponders container is present

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L136

    This attribute is MANDATORY and cannot be profiled
    
    **Proposal**: make it optional, clarifying that if not present, the information is not reported by the server (profile)
    
    NEW
    ```
           +--ro equalization-mode?                      identityref
    ```

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L143

    How to interpret an empty list?
    
    **Proposal**: use a presence container to indicate whether OMS-elements are reported
    
    NEW
    ```
           +--ro media-channel-groups!
              +--ro media-channel-group* [i]
    ```

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L214-L216

    **Proposal**: add min-element 1 and a when statement based on the presence of the transponders container

1. - [x] https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/blob/5fc9e2dbabc73202871b65b06fb18a3153d297bc/ietf-optical-impairment-topology.tree#L224-L225

    No ambiguity when this attribute is not present:
    - if protection is not enabled, the value 'te-types:lsp-protection-unprotected' is reported;
    - it this attribute is not present, the server does not report the protection configuration (profile)

See also: [oi-profile-comments-00.txt](https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/files/10796322/oi-profile-comments-00.txt)

Note that some of the proposed changes to YANG are partially addressing also issue #110

### PR DESCRIPTION
**Pull request by italobusi**
Updates for IETF 116:
- Change the name of nmc-path-id: fix #125
- Remove key from media channel list: see #124
- Updated to allow profiling the OI model: fix #126
- Updated Victor mail address: supersede #82 

Pending merge of PR https://github.com/ietf-ccamp-wg/ietf-ccamp-layer0-types-ext-RFC9093-bis/pull/67

Co-Authored-By: sergio belotti <sergio.belotti@nokia.com>

### CODE REVIEW

**Review by EstherLerouzic — COMMENTED** [review]
Looks good to me, except for one description that I do not understand.
I think that the json example must be changed accordingly, right ?

File: ietf-optical-impairment-topology.yang [hunk 1]
```diff
@@ -703,129 +709,146 @@ module ietf-optical-impairment-topology {
...
+              path "/nw:networks/nw:network/otsi-information/" +
+                  "otsi-group/otsi-group-id";
             }
             description
-              "Reference to the OTSi within the OTSiG carried
-              by this media channel.";
+              "Reference to the otsi-group list to get otsi-group
+              identifier of the
+              OTSiG carried by this media channel 
+              that reports the transient stat";
```
[the remark this answers is not in the document]
Comment (italobusi) [reply ietf-optical-impairment-topology.yang]: It seems an old cut&paste mistake (the text is already present in the published draft)

Anyhow description fixed in https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/pull/129/commits/327b2540b374973dbfbefdc44610284043b80e06

@EstherLerouzic is the new description fine?

**Comment by italobusi** [comment]
> I think that the json example must be changed accordingly, right ?

Yes ... we will try to fix them before the submission deadline

**Review by aguoietf — APPROVED** [review]
Changes look good to me.

**Comment by italobusi** [comment]
> 

Just done and checked with yanglint

@EstherLerouzic could you please double check with your automatic tools?

**Comment by EstherLerouzic** [comment]
> > 
> 
> Just done and checked with yanglint
> 
> @EstherLerouzic could you please double check with your automatic tools?

OK for me !

**Pull request merged**